main.c

```diff
@@ -324,24 +322,34 @@ static void keyboard_handle_key(void *data, struct wl_keyboard *wl_keyboard,
 			}
 			state->edit_anchor = true;
 			break;
+		
 		case XKB_KEY_Shift_L:
 		case XKB_KEY_Shift_R:
 			if (!state->fixed_aspect_ratio) {
 				state->aspect_ratio = 1;
 				recompute_selection(seat);
 			}
 			break;
+
+		case XKB_KEY_Return:
```

Comment:
What if return is hit while you are currently dragging a selection? Should we then store the current selection in the result?